### PR TITLE
Open in new tab button

### PR DESCRIPTION
**Pull request by reteps**
Formerly #202.

### CODE REVIEW

**Comment by aopell** [comment]
@reteps can you explain the utility of this change and what exactly it does?

**Comment by reteps** [comment]
image
@aopell It adds a button to open the iframe content in a new tab.

**Review by glen3b — CHANGES_REQUESTED** [review]
Thanks for the PR! I've highlighted a couple more specific/implementation-level changes in my comments I'm hoping you can take a look at; I don't expect any of them to require huge reworks.

My biggest design-level concern is the placement of the button. I don't like how it's out-of-the-way and hard-to-see when looking at the content frame. Could we perhaps move it closer to the content, like just to the right of the iframe? We also may want a more visual and less textual indication of the action.

@aopell may have more insight on the design front.

**Comment by reteps** [comment]
image

File: js/page.js
Comment (glen3b) [inline js/page.js:1]: We should probably leave a one-line comment above this function saying what it does, e.g.
```js
// Adds an "Open in New Tab" link to /page pages which simply embed something.
(function () {
...
```

Especially since we have early returns from this function, so we wouldn't want to inadvertently add unrelated changes in the same anonymous function.

File: js/page.js
Comment (glen3b) [inline js/page.js:6]: Per our discussion on Discord, we'll probably take another pass at the styling here. My first intuition is that we probably won't make the entire title a link, just the button; we'll probably also change the icon. That said, nice work!

File: js/page.js
Comment (glen3b) [inline js/page.js:23]: I'm still not super keen on the matching/replacement logic here. Matching specifically on `pub?embedded` as a string literal seems very fragile (what if there's another query parameter?) - I'd be okay with stripping the `embedded` parameter from any GDocs URL.

I'm also okay with eschewing hard-to-read regexes - as I mentioned in another comment, there should be a URL parsing API you can use. Once you use that, semantically meaningful query string manipulation should be fairly straightforward (or, at worst, doable with a not-bad regex, although I'd be shocked if there isn't a builtin component for doing almost exactly what we want). Likewise, if you can have the browser isolate the path component of the URL for you, using something like
```js
pathComponent.replace(/preview(\/?)$/, "/edit$1")
```
should be fairly trivial.

**Review by glen3b — CHANGES_REQUESTED** [review]
I've just got a couple nits left, mostly relating to the URL reprocessing logic. Since, in theory (I believe), _any_ URL can be embedded, I'd really like that replacement logic to be solid. That's pretty much it from my end though.

If you're having trouble with the replacement logic or if it ends up looking like it'll be complex, then we can revisit it - I don't want that function getting unwieldy in the name of purity.

**Comment by glen3b** [comment]
Actually, one more comment: a Google Slides embedded page I found has the following URL:
`https://docs.google.com/presentation/d/e/SOME-GDOCS-ID/embed?start=false&loop=false&delayms=3000#slide=SOME-ID`.

I'm not sure if there's a better URL to use for "open in new tab" purposes (nothing obvious came up in my quick testing), but I figured I'd note down this case.

File: js/page.js
Comment (aopell) [reply js/page.js:23]: Yeah, I believe [URL.pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname) is what you're looking for here

**Review by aopell — APPROVED** [review]
As far as implementation goes, I'm going to defer to @glen3b, but for design I think this is definitely an improvement over the sidebar button. I'll take a look at making a better icon once this PR is merged.

File: js/page.js
Comment (glen3b) [inline js/page.js:23]: What is the significance of the `s`?

File: js/page.js
Comment (glen3b) [inline js/page.js:25]: Missing semicolon. Also, I'm not sure if it matters, but if there's a trailing slash in the original URL, it will not be preserved in the replacement.

File: js/page.js
Comment (glen3b) [inline js/page.js:27]: Missing semicolon. Also I'm not sure if this is resilient against all cases:
- embedded is the first and only parameter
- embedded is the first of multiple parameters
- embedded is a middle parameter (3+)
- embedded is the last of multiple parameters

Comment (glen3b) [reply js/page.js:27]: If there's a way to parse query strings that might be simpler.

**Review by glen3b — CHANGES_REQUESTED** [review]
It's just nits at this point.

File: js/page.js
Comment (reteps) [reply js/page.js:25]: does not matter

**Review by glen3b — APPROVED** [review]
Looks great! Thanks for your patience.

**Comment by reteps** [comment]
Umm i need another fix for `htmlembed`

**Comment by aopell** [comment]
> Umm i need another fix for `htmlembed`

no problem, just open a new PR

**Pull request merged**
